cmds/tui.c

```diff
@@ -431,22 +433,29 @@ static int build_tui_node(struct uftrace_task_reader *task,
 
 		name = symbol_getname(sym, addr);
 
-		if (rec->type == UFTRACE_EXIT)
-			update_report_node(task, name, tg);
+		if (rec->type == UFTRACE_EXIT) {
+			if (opts->srcline)
+				loc = task_find_loc_addr(&task->h->sessions, task, rec->time, addr);
+
+			update_report_node(task, name, tg, loc);
+		}
 	}
 	else if (rec->type == UFTRACE_EVENT) {
 		sym = &sched_sym;
 		name = symbol_getname(sym, addr);
 
 		if (addr == EVENT_ID_PERF_SCHED_IN)
-			update_report_node(task, name, tg);
+			update_report_node(task, name, tg, NULL);
 		else if (addr != EVENT_ID_PERF_SCHED_OUT)
 			return 0;
 	}
 	else  /* rec->type == UFTRACE_LOST */
 		return 0;
 
-	graph_add_node(tg, rec->type, name, sizeof(struct tui_graph_node));
+	if (opts->srcline && loc == (void*)-1)
```

Comment:
I know it's weird. it's used to call task_find_loc_addr once by checking 1 variable.
could you give me your opinion?